dcos/dcos-ui
# DCOS-48322: Jobs container form section

This PR adds a container form section to the jobs form. See intended designs [here](https://mesosphere.invisionapp.com/share/Y6Q8EFKBA3G#/screens/346213421).

## Testing

Try testing the form validation and editing via JSON editor. Some validation scenarios:

Container section disabled:
- In General tab, choose Command Only 
- In Container tab, all form fields should be disabled with an inline banner at the top

Sharing common attributes when switching between UCR and docker
- In General tab, choose Container Image
- Enter something in Container Image text input - JSON editor should show a `ucr` object with an `image: { id: "what you entered" }`
- Go to Container tab, switch to Docker - JSON editor should now show a `docker` object with `image: "what you entered"`
- With docker still chosen, check "Force Pull Image On Launch" - JSON editor should add `forcePullImage: true` to `docker` object
- Switch to UCR - JSON editor should show `ucr` object with `image: { ..., forcePull: true }`

Docker parameters
- Choose Docker
- Add parameters with JSON editor open - they should only appear in job JSON if they have a key or value filled out
- Adding parameters should add `parameters: [{ key: "x", value: "y" }]` to `docker` object
- If you submit the form when a docker parameter has either an empty key or empty value, you should get an error

Args
- Choose docker
- Add args with JSON editor open - they should only appear in job JSON if they are not empty
- Adding args should add `args: ["x", "y", ...]` to `run` object
- If you add `args` manually in JSON editor with UCR chosen and then submit, you should get an error
- If you submit the form with both Args and a Command specified, you should get an error

## Trade-offs

Building on patterns established in #3599 

## Dependencies

N/A
